csharp/src/Apache.Arrow/Schema.cs

```diff
@@ -63,5 +63,48 @@ public int GetFieldIndex(string name, StringComparer comparer = default)
             return _fields.IndexOf(
                 _fields.Single(x => comparer.Compare(x.Name, name) == 0));
         }
+
+        public override bool Equals(object obj)
```

Comment:
There's a few more things you should do when you override Equals in .NET.

- Override and implement `GetHashCode()`
- Also implement `IEquatable<T>`.

See [https://docs.microsoft.com/en-us/visualstudio/ide/reference/generate-equals-gethashcode-methods?view=vs-2017](https://docs.microsoft.com/en-us/visualstudio/ide/reference/generate-equals-gethashcode-methods?view=vs-2017) for what VS will automatically generate for you.